src/torcontrol.cpp

```diff
@@ -524,11 +524,15 @@ void TorController::auth_cb(TorControlConnection& _conn, const TorControlReply&
 
         // Now that we know Tor is running setup the proxy for onion addresses
         // if -onion isn't set to something else.
-        if (gArgs.GetArg("-onion", "") == "") {
+        // Or if we have already an outbound proxy defined. we skip this footgun gracefully.
+        proxyType addrOnion;
+        if (gArgs.GetArg("-onion", "") == "" && !GetProxy(NET_ONION, addrOnion)) {
             CService resolved(LookupNumeric("127.0.0.1", 9050));
-            proxyType addrOnion = proxyType(resolved, true);
-            SetProxy(NET_ONION, addrOnion);
-            SetReachable(NET_ONION, true);
+            addrOnion = proxyType(resolved, gArgs.GetBoolArg("-proxyrandomize", true));
+            if (SetProxy(NET_ONION, addrOnion)) {
+                LogPrintf("tor: Default outbound tor .onion address proxy is set to 127.0.0.1:9050\n");
+                SetReachable(NET_ONION, true);
+            }
```

Comment:
I think this should also log when `SetProxy` fails, as It means there is a bug in the software. After all it tries to set a hardcoded value which *should* never fail.